dev-util/gn/patches/gn-091169b.patchset

```diff
@@ -0,0 +1,404 @@
+index 61239ad..e5cb6a1 100644
+--- a/base/files/file.h
++++ b/base/files/file.h
+@@ -22,10 +22,10 @@
+ 
+ namespace base {
+ 
+-#if defined(OS_BSD) || defined(OS_MACOSX) || defined(OS_NACL) || \
++#if defined(OS_BSD) || defined(OS_MACOSX) || defined(OS_NACL) || defined(OS_HAIKU) || \
+     defined(OS_ANDROID) && __ANDROID_API__ < 21
+ typedef struct stat stat_wrapper_t;
+-#elif defined(OS_POSIX) || defined(OS_FUCHSIA)
++#elif defined(OS_POSIX) && !defined(OS_HAIKU) || defined(OS_FUCHSIA)
+ typedef struct stat64 stat_wrapper_t;
+ #endif
+ 
+diff --git a/base/files/file_posix.cc b/base/files/file_posix.cc
+index 4faf77b..62bb322 100644
+--- a/base/files/file_posix.cc
++++ b/base/files/file_posix.cc
+@@ -24,7 +24,7 @@ static_assert(File::FROM_BEGIN == SEEK_SET && File::FROM_CURRENT == SEEK_CUR &&
+ 
+ namespace {
+ 
+-#if defined(OS_BSD) || defined(OS_MACOSX) || defined(OS_NACL) || \
++#if defined(OS_BSD) || defined(OS_MACOSX) || defined(OS_NACL) || defined(OS_HAIKU) || \
+     defined(OS_ANDROID) && __ANDROID_API__ < 21
+ int CallFstat(int fd, stat_wrapper_t* sb) {
+   return fstat(fd, sb);
+@@ -100,7 +100,7 @@ void File::Info::FromStat(const stat_wrapper_t& stat_info) {
+   int64_t last_accessed_nsec = stat_info.st_atimespec.tv_nsec;
+   time_t creation_time_sec = stat_info.st_ctimespec.tv_sec;
+   int64_t creation_time_nsec = stat_info.st_ctimespec.tv_nsec;
+-#elif defined(OS_AIX)
++#elif defined(OS_AIX) ||  defined(OS_HAIKU)
+   time_t last_modified_sec = stat_info.st_mtime;
+   int64_t last_modified_nsec = 0;
+   time_t last_accessed_sec = stat_info.st_atime;
+diff --git a/base/files/file_util_posix.cc b/base/files/file_util_posix.cc
+index 23c7cb4..49b4f31 100644
+--- a/base/files/file_util_posix.cc
++++ b/base/files/file_util_posix.cc
+@@ -57,7 +57,7 @@ namespace base {
+ 
+ namespace {
+ 
+-#if defined(OS_BSD) || defined(OS_MACOSX) || defined(OS_NACL) || \
++#if defined(OS_BSD) || defined(OS_MACOSX) || defined(OS_NACL) || defined(OS_HAIKU) || \
+     defined(OS_ANDROID) && __ANDROID_API__ < 21
+ int CallStat(const char* path, stat_wrapper_t* sb) {
+   return stat(path, sb);
+diff --git a/base/strings/sys_string_conversions_posix.cc b/base/strings/sys_string_conversions_posix.cc
+index 0e14428..113a50a 100644
+--- a/base/strings/sys_string_conversions_posix.cc
++++ b/base/strings/sys_string_conversions_posix.cc
+@@ -26,7 +26,7 @@ std::wstring SysUTF8ToWide(StringPiece utf8) {
+   return out;
+ }
+ 
+-#if defined(SYSTEM_NATIVE_UTF8) || defined(OS_ANDROID)
++#if defined(SYSTEM_NATIVE_UTF8) && !defined(OS_HAIKU) || defined(OS_ANDROID)
+ // TODO(port): Consider reverting the OS_ANDROID when we have wcrtomb()
+ // support and a better understanding of what calls these routines.
+ 
+@@ -158,6 +158,6 @@ std::wstring SysNativeMBToWide(StringPiece native_mb) {
+   return out;
+ }
+ 
+-#endif  // defined(SYSTEM_NATIVE_UTF8) || defined(OS_ANDROID)
++#endif  // defined(SYSTEM_NATIVE_UTF8) || defined(OS_ANDROID) || !defined(OS_HAIKU)
```

Comment:
```diff
-+#endif  // defined(SYSTEM_NATIVE_UTF8) || defined(OS_ANDROID) || !defined(OS_HAIKU)
++#endif  // defined(SYSTEM_NATIVE_UTF8) && !defined(OS_HAIKU) || defined(OS_ANDROID)
```

Reply:
@fbrosson Resolved in newer patchset.

Reply:
:+1:

BTW, please forgive me for not merging your PR. I haven't tested the recipe (since a while) so I'll let someone else do the merge.